### PR TITLE
LPC entropy

### PR DESCRIPTION
**Pull request by agansari**
Adds new RNG device for NXP's LPC family and enable it on LPC55S69 devices.

Depends on: https://github.com/zephyrproject-rtos/hal_nxp/pull/40

### CODE REVIEW

File: dts/bindings/rng/nxp,lpc-rng.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2018, Linaro Limited
```
Comment (galak) [inline dts/bindings/rng/nxp,lpc-rng.yaml:1]: 2020

File: dts/bindings/rng/nxp,lpc-rng.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+description: LPC RNG (True Random Number Generator)
```
Comment (galak) [inline dts/bindings/rng/nxp,lpc-rng.yaml:4]: Drop `True`

**Review by galak — CHANGES_REQUESTED** [review]
Should you add this to the LPC55S16 as well? (@henrikbrixandersen?)

**Review by galak — COMMENTED** [review]
I'd merge down the commits.  Seems like one for the driver, binding and SoC DTSI updates would be fine.

File: drivers/entropy/entropy_mcux_rng.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	__ASSERT_NO_MSG(!status);
+
+	return 0;
+}
+
+static const struct entropy_driver_api entropy_mcux_rng_api_funcs = {
+	.get_entropy = entropy_mcux_rng_get_entropy
+};
+
+static struct mcux_entropy_config entropy_mcux_config = {
```
Comment (MaureenHelm) [inline drivers/entropy/entropy_mcux_rng.c:38]: add `const`

**Comment by henrikbrixandersen** [comment]
> Should you add this to the LPC55S16 as well? (@henrikbrixandersen?)

The LPC55S1x has the RNG at offset 0x3a000 as well. I have not tested it.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/rng/nxp,lpc-rng.yaml (hunk 2 above)
Comment (agansari) [reply dts/bindings/rng/nxp,lpc-rng.yaml:4]: @galak dropping "True" as it's confusing, the user manual names the RNG device as a True RNG.

**Comment by agansari** [comment]
@galak @MaureenHelm please revisit this PR.
@henrikbrixandersen i've added the device to lpc55s16 as well.

**Comment by henrikbrixandersen** [comment]
Do we have test case I can use to verify it on the LPCXpresso55S16?

**Comment by agansari** [comment]
> Do we have test case I can use to verify it on the LPCXpresso55S16?

I've ran `west build -b lpcxpresso55s69_cpu0 tests/crypto/rand32 --pristine` and it passes.
I've built for lpcxpresso55s16_ns, SW RNG is disabled, hal_nxp functions are linked. Can't test as I don't have a board.

File: west.yml [hunk 4]
```diff
@@ -92,7 +92,7 @@ manifest:
       revision: 1c4fdba512b268033a4cf926bddd323866c3261a
       path: tools/net-tools
     - name: hal_nxp
-      revision: 39615db4095858d24d87dfd6d8e0970704aae18a
+      revision: pull/40/head
```
Comment (henrikbrixandersen) [inline west.yml:95]: This needs to be updated before merging.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Verified on `lpcxpresso55s16_ns`. The nxp_hal reference needs to be updated before merging.

File: west.yml (hunk 4 above)
Comment (MaureenHelm) [reply west.yml:95]: The hal pr has now been merged, so you can update this to the new SHA

Comment (agansari) [reply west.yml]: @MaureenHelm  @henrikbrixandersen i've updated west.yaml with the commit number.

Comment (MaureenHelm) [reply west.yml]: Please update the commit message too. `Updated to west.yaml pull 40 on hal_nxp repository.` doesn't make sense anymore

**Review by henrikbrixandersen — APPROVED** [review]

File: west.yml (hunk 4 above)
Comment (agansari) [reply west.yml]: changed commit message

Comment (MaureenHelm) [reply west.yml]: Sorry to be a pain, but the commit message still isn't right - it's referring to the old hal commit. You could change it to `Updated west.yaml to the hal_nxp commit which enabled entropy drivers for LPC family.`

Comment (agansari) [reply west.yml]: @MaureenHelm done!

**Review by MaureenHelm — APPROVED** [review]

File: west.yml (hunk 4 above)
Comment (MaureenHelm) [reply west.yml]: thank you!

**Review by galak — APPROVED** [review]

**Pull request merged**